src/ui/public/vis_maps/lib/tilemap_settings.js

```diff
@@ -149,13 +149,39 @@ uiModules.get('kibana')
        * Get the options of the default TMS
        * @return {{}}
        */
-      getOptions() {
+      getTMSOptions() {
         if (!this._settingsInitialized) {
           throw new Error('Cannot retrieve options before calling .loadSettings first');
         }
-        return this._options;
+        return this._tmsOptions;
       }
 
+
+      /**
+       * @return {{maxZoom: (*|number), minZoom: (*|number)}}
+       */
+      getMinMaxZoom(isWMSEnabled) {
```

Comment:
I think this belongs in the src/ui/public/vis_maps/visualizations/_map.js where is will know if wms is enabled. the `TilemapSettings` shouldn't be concerned with WMS at all